src/utils_cmd_putval.c

```diff
@@ -49,9 +58,29 @@ static int set_option(value_list_t *vl, const char *key, const char *value) {
 
     if ((errno == 0) && (endptr != NULL) && (endptr != value) && (tmp > 0.0))
       vl->interval = DOUBLE_TO_CDTIME_T(tmp);
-  } else
-    return 1;
+  } else if (strncasecmp("meta:", key, 5) == 0) {
+    const char *meta_key = key + 5;
+    size_t value_len = strlen(value);
+
+    if (vl->meta == NULL) {
+      vl->meta = meta_data_create();
+      if (vl->meta == NULL) {
+        return 1;
+      }
+    }
 
+    if (is_quoted(value, value_len)) {
+      const char *value_str = strndup(value + 1, value_len - 2);
+      if (value_str == NULL) {
+        return 1;
+      }
+      meta_data_add_string(vl->meta, meta_key, value_str);
+      free((void *)value_str);
```

Comment:
This is missing a return 0, right? Also, I think we should print an error in the 'else' case:

    if (is_quoted(value, value_len)) {
      …
      return 0;
    }
    cmd_error(CMD_PARSE_ERROR, err, "Non-string metadata not supported yet");
    return CMD_PARSE_ERROR;

Reply:
Hrm, also, the caller currently ignores the return value of set_option, so that would have to be fixed as well :-/

Reply:
Done. I am using the values defined in utils_cmds.h [1] here.

[1]: https://github.com/collectd/collectd/blob/9ae06bc2026e041e937828379daa232d3c0afbf7/src/utils_cmds.h#L111